### PR TITLE
Translator for globes

### PR DESCRIPTION
**Pull request by eranroz**
A custom translator for globes, for properly extracting the author

### CODE REVIEW

File: Globes.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+function doWeb(doc, url) { 
+	var item = new Zotero.Item('newspaperArticle');
+	
+	item.title = ZU.xpathText(doc, '//meta[@property="og:title"]/@content');
+	item.publicationTitle = 'Globes';
+	item.url = url;
+	item.language = 'he';
+	var abstract = ZU.xpathText(doc, '//meta[@name="description"]/@content');
+	if (abstract) {
+		item.abstractNote = abstract;
```
Comment (zuphilip) [inline Globes.js]: You can assign this also directly, if no postprocessing is needed.

File: Globes.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	}
+	
+	var json = ZU.xpathText(doc, '//script[@type="application/ld+json"]');
+	var data = JSON.parse(json);
+	if (data) {
+		
+		if (data.author && Array.isArray(data.author)) {
+			// note only the 0 item is a real author
+			item.creators.push(Zotero.Utilities.cleanAuthor(data.author[0].name, 'author'));
```
Comment (zuphilip) [inline Globes.js]: How about articles with multiple authors? I guess we should restrict this in another way...

File: Globes.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+				"date": "2019-01-10",
+				"abstractNote": "גורמים עלומים בצפון-מזרח תאילנד מנהלים כבר שנים מערך מתוחכם ומורכב להפליא של ציד בלתי חוקיים, ומרוויחים הון עתק מהכליה שהם ממיטים על כמה מהמינים האקזוטיים בעולם ● האם האדם היחיד שנאבק בתופעה יצליח להפיל את סינדיקט ההברחות שאיש לא יכול לו?&nbsp;●&nbsp;גיליון שכולו קנאה&nbsp;&nbsp;",
+				"language": "he",
+				"libraryCatalog": "Globes",
+				"publicationTitle": "Globes",
+				"url": "https://www.globes.co.il/news/article.aspx?did=1001302127",
+				"attachments": [],
+				"tags": [],
+				"notes": [],
+				"seeAlso": []
```
Comment (zuphilip) [inline Globes.js]: Can we save section within the newspaper where this article appeared as well? I.e.
```html
<meta property="article:section" content="מגזין G">
```

File: Globes.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+					}
+				],
+				"date": "2019-01-10",
+				"abstractNote": "גורמים עלומים בצפון-מזרח תאילנד מנהלים כבר שנים מערך מתוחכם ומורכב להפליא של ציד בלתי חוקיים, ומרוויחים הון עתק מהכליה שהם ממיטים על כמה מהמינים האקזוטיים בעולם ● האם האדם היחיד שנאבק בתופעה יצליח להפיל את סינדיקט ההברחות שאיש לא יכול לו?&nbsp;●&nbsp;גיליון שכולו קנאה&nbsp;&nbsp;",
+				"language": "he",
+				"libraryCatalog": "Globes",
+				"publicationTitle": "Globes",
+				"url": "https://www.globes.co.il/news/article.aspx?did=1001302127",
+				"attachments": [],
+				"tags": [],
```
Comment (zuphilip) [inline Globes.js]: Can we add some tags as well? Maybe from here
```html
<meta property="article:tag" content="גיליון שכולו קנאה">
```

File: Globes.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+						"creatorType": "author"
+					}
+				],
+				"date": "2019-01-10",
+				"abstractNote": "גורמים עלומים בצפון-מזרח תאילנד מנהלים כבר שנים מערך מתוחכם ומורכב להפליא של ציד בלתי חוקיים, ומרוויחים הון עתק מהכליה שהם ממיטים על כמה מהמינים האקזוטיים בעולם ● האם האדם היחיד שנאבק בתופעה יצליח להפיל את סינדיקט ההברחות שאיש לא יכול לו?&nbsp;●&nbsp;גיליון שכולו קנאה&nbsp;&nbsp;",
+				"language": "he",
+				"libraryCatalog": "Globes",
+				"publicationTitle": "Globes",
+				"url": "https://www.globes.co.il/news/article.aspx?did=1001302127",
+				"attachments": [],
```
Comment (zuphilip) [inline Globes.js]: Let us also save a snapshot, i.e. something like
```js
			item.attachments.push({
				title: "Snapshot",
				document: doc,
				mimeType:"text/html"
			});
```

File: Globes.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+function detectWeb(doc, url) {
+	return 'newspaperArticle';
+}
+function doWeb(doc, url) { 
+	var item = new Zotero.Item('newspaperArticle');
+	
+	item.title = ZU.xpathText(doc, '//meta[@property="og:title"]/@content');
+	item.publicationTitle = 'Globes';
+	item.url = url;
+	item.language = 'he';
```
Comment (zuphilip) [inline Globes.js]: Can you also handle the English version of the Globes, e.g. https://en.globes.co.il/en/article-will-competition-authority-approve-taboola-outbrain-merger-1001302929 ?

File: Globes.js (hunk 3 above)
Comment (zuphilip) [reply Globes.js]: This then should go into the field `section`.

File: Globes.js (hunk 2 above)
Comment (eranroz) [reply Globes.js]: added to test cases few different cases - one where multiple "authors" are real author, and date and one where there are really 2 authors.

File: Globes.js (hunk 3 above)
Comment (eranroz) [reply Globes.js]: Done

File: Globes.js (hunk 4 above)
Comment (eranroz) [reply Globes.js]: Done

File: Globes.js (hunk 5 above)
Comment (eranroz) [reply Globes.js]: Done

File: Globes.js (hunk 6 above)
Comment (eranroz) [reply Globes.js]: Done

File: Globes.js [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+				"creators": [
+					{
+						"firstName": "ג'ושוע",
+						"lastName": "האמר",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "",
+						"lastName": "GQ",
+						"creatorType": "author"
```
Comment (zuphilip) [inline Globes.js]: Is this a real author or should we rather filter this one out?

Comment (eranroz) [reply Globes.js]: this a translated article (original aticle:  https://www.gq.com/story/hunting-rhino-horn-cartel-of-thailand ) and the credits are to GQ. 
I'm not sure how to determine it programmatically (other than heuristics of having both firstName and lastName?)

File: Globes.js [hunk 8]
```diff
@@ -0,0 +1,248 @@
+{
+	"translatorID": "4e9c34ef-89bb-4c7e-97dd-abb61d8b9254",
+	"label": "Globes",
+	"creator": "Eran Rosenthal",
+	"target": "^https?://www\\.globes\\.co\\.il/news/article.aspx",
```
[the remark this answers is not in the document]
Comment (zuphilip) [reply Globes.js]: @eranroz That seems still open.

File: Globes.js (hunk 7 above)
Comment (zuphilip) [reply Globes.js]: I would suggest only to not take it as author if it is exactly `GQ`. (If we have more cases like this this, then we may see better about which pattern might be suitable.)

File: Globes.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (eranroz) [reply Globes.js]: Fixed

File: Globes.js (hunk 7 above)
Comment (eranroz) [reply Globes.js]: Seems the site changes the metadata in that item - updating the test itself

**Comment by zuphilip** [comment]
Great! I did a small improvement in the detection and merge this now! :rocket:

Thank you very much @eranroz !

**Pull request merged**